Update/theme collection desktop style

dsas opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/4072

## Proposed Changes

Theme collections: Improve desktop styling
    
 * Used font sizes from designs (but not font family until we're ready to do the whole page at once)
 * Removed unnecessary vertical space between headings and carousel
 * Limited width of subheading text
 * Removed some unnecessary divs
 * Tweaked navigation buttons

## Testing Instructions
 1. Use calypso live link
 2. In a logged-out window visit /themes?flags=themes/discovery-lots

Before | After
-------|------
<img width="1450" alt="Screenshot 2023-10-03 at 13 59 46" src="https://github.com/Automattic/wp-calypso/assets/93301/1253b373-78e9-4f7e-ae6b-881b255f37bb"> | <img width="1450" alt="Screenshot 2023-10-03 at 13 57 41" src="https://github.com/Automattic/wp-calypso/assets/93301/407eb526-e23f-4491-8404-684ba6a19c39">



<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

dsas committed:
Theme collections: Improve desktop styling

 * Used font sizes from designs
 * Removed unnecessary vertical space between headings and carousel
 * Limited width of subheading text
 * Removed some unnecessary divs
 * Tweaked navigation buttons

More styling will come later as this is made mobile friendly, and as
more design work gets finalised.

rcrdortiz commented:
I'm testing using the container, I loaded this URL in an incognito window: https://container-infallible-fermi.calypso.live/themes?flags=themes/discover-lots 
But I can't seem to get the new feature to load, I'm sure that I have something completely wrong but can't figure out what it is 🤔

dsas commented:
> I'm testing using the container, I can't seem to get the new feature to load, 

Sorry @rcrdortiz, this is all down to me, I had the wrong flag name, it should be themes/discover**y**-lots (I updated the initial post)

rcrdortiz commented:
> > I'm testing using the container, I can't seem to get the new feature to load,
> 
> Sorry @rcrdortiz, this is all down to me, I had the wrong flag name, it should be themes/discover**y**-lots (I updated the initial post)

Oh, wow I didn't notice the typo either! Thanks for updating.

rcrdortiz commented:
I'm not sure if this is the right PR for this so feel free to move this suggestion to a new issue/s if we decide to move forward with it. Besides these things the code looks good and works as expected.

### Disable state instead of removed
After scrolling to the end of the carousel the next button disappears, maybe we could have the next (and or prev) button in a disabled state instead WDYT?

This is how it looks now:
![Captura de pantalla 2023-10-04 a las 11 07 21](https://github.com/Automattic/wp-calypso/assets/1989914/6bb867c2-7058-43f7-8bcf-0848be4895a2)

What I have in mind (we can adjust the opacity or whatever we think best suits this specific use case):
![Captura de pantalla 2023-10-04 a las 11 09 03](https://github.com/Automattic/wp-calypso/assets/1989914/c99a1769-19b6-45f2-abd1-a1aa6d68dd0e)

### Fade out animation isn't as smooth as fade in
Another thing I noticed is that the animation for showing the next/prev arrows is much smoother than when we hide the arrows. Could we use the same easing on both animations?

https://github.com/Automattic/wp-calypso/assets/1989914/d68ddc85-f7b9-4dfd-a6f3-392de4ae70f7

### Carousel position after returning
When we return from the theme preview the carousel position is reset and I need to scroll all the way to my last position.

dsas commented:
> When we return from the theme preview the carousel position is reset and I need to scroll all the way to my last position.

This one is kinda separate. For some reason (maybe to do with our routing?) it doesn't Just Work out of the box as advertised. I can make it update the URL, I haven't yet made it read from the URL and change slides. I've pushed the [`try/discovery-themes-history` branch](https://github.com/Automattic/wp-calypso/compare/trunk...try/discovery-themes-history), but I'm just going to leave it parked for now, I think it is something to come back to later.

rcrdortiz commented:
The new changes look ❤️ now!

I noticed we could use a bit more spacing between the first carousel and the header/hero of the LoTS. The header seems to be close, WDTY? Besides that, everything looks great now. I also re-triggered the desktop and mobile E2E that were failing, if they fail again it might be something we need to look into.

![Captura de pantalla 2023-10-05 a las 15 48 09](https://github.com/Automattic/wp-calypso/assets/1989914/6b9507b2-8d94-4133-9468-cef022aee211)

rcrdortiz commented:
Perfect now 🥇 
🚢 🚀

dsas commented:
Just to note we have some updated figma designs: RcxiuWYjZGkFRdlWL7L02m-fi-1240%3A25645 I'll get this merged as-is and then come back and update to match that in a separate PR.

dsas committed:
Fix padding on upsell nudge

The padding has been moved which meant there was no longer any padding
between the upsell nudge and the content below it.
